package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -90,6 +91,47 @@ def unique_int_1d(np.intp_t[:] values):
     return np.array(result)
 
 
+@cython.boundscheck(False)
+def _in2d(np.intp_t[:, :] arr1, np.intp_t[:, :] arr2):
+    """Similar to np.in1d except works on 2d arrays
+
+    Parameters
+    ----------
+    arr1, arr2 : numpy.ndarray, shape (n,2) and (m, 2)
+       arrays of integers
+
+    Returns
+    -------
+    in1d : bool array
+      if an element of arr1 was in arr2
+
+    .. versionadded:: 1.1.0
+    """
+    cdef object out
+    cdef ssize_t i
+    cdef cset[pair[np.intp_t, np.intp_t]] hits
+    cdef pair[np.intp_t, np.intp_t] p
+
+    if not arr1.shape[1] == 2 or not arr2.shape[1] == 2:
+        raise ValueError("Both arrays must be (n, 2) arrays")
+
+    for i in range(arr2.shape[0]):
+        p = pair[np.intp_t, np.intp_t](arr2[i, 0], arr2[i, 1])
+        hits.insert(p)
```

Comment:
Not sure if we need a comment or two about the C++ magic happening here; I suppose we are slowly growing expertise with devs familiar with C++ (I'm not one of them).